doc/developers.rst

```diff
@@ -451,6 +444,138 @@ DataAccess class i.e. the `_field_write_arguments()` and
 `Arguments` class.
 
 
+Parsing Code
+############
+
+The PSyclone `parse` module is responsible for parsing science
+(algorithm and kernel) code and extracting the required information
+for the algorithm translation and PSy generation phases.
+
+The `parse` module contains modules for parsing algorithm
+(`algorithm.py`) and kernel (`kernel.py`) code as well as a utility
+module (`utils.py`) for common functionality.
+
+Parsing Algorithm Code
+======================
+
+The first thing PSyclone typically does is parse an input file. The
+code to do this is found in `parse/algorithm.py`.
+
+An input file can be parsed via the `parse` function or via an
+instance of the `Parser` class. In practice the `parse` function
+simply calls the `Parser` class so we will concentrate on the latter
+in this section. The `parse` function could be removed from PSyclone
+but it is simple and is used in existing PSyclone scripts and
+examples.
+
+The `Parser` class is initialised with a number of optional
+arguments. A particular `api` can be specified (this is required so
+the parser knows what sort of code and metadata to expect, how to
+parse it and which `builtins` are supported). The name used to specify
+an invoke (defaulting to `invoke`) can be changed, the name of the
+module supplying any infrastructure (defaulting to `inf`) can be
+changed, a path to where to look for associated kernel files can be
+provided and a particular maximum line length can be specified.
+
```

Comment:
The joy of documentation is that I read this and thought, what's 'inf' ever used for and the answer is: nothing. Please could you therefore remove it from the `Parser` constructor. You could also just use Sphinx's autodoc feature here and bring in the class documentation from the source.

Reply:
Good point. Done.